tests/integration/helpers.py

```diff
@@ -21,11 +22,95 @@
 APP_NAME_SERVER = "temporal-k8s"
 APP_NAME_ADMIN = "temporal-admin-k8s"
 
-WORKER_CONFIG = {
+BASE_WORKER_CONFIG = {
     "namespace": "default",
     "queue": "test-queue",
 }
 
+SECRETS_WITH_VAULT_CONFIG = dedent(
+    """
+        secrets:
+            env:
+                - key1: value1
+                - key2: value2
+            vault:
+                - path: vault-secrets
+                  key: vault-secret1
+                - path: vault-secrets
+                  key: vault-secret2
+        """
+)
+
+
+def get_worker_config(secret_id):
+    """Get worker charm config.
+
+    Args:
+        secret_id: Juju secret id.
+
+    Returns:
+        Temporal worker charm config
+    """
+    return {
+        **BASE_WORKER_CONFIG,
+        "secrets": dedent(
+            f"""
+        secrets:
+            env:
+                - message: hello
+                - key2: value2
+            juju:
+                - secret-id: {secret_id}
+                  key: juju-secret1
+                - secret-id: {secret_id}
+                  key: juju-secret2
+        """
+        ),
+    }
+
+
+async def add_juju_secret(ops_test: OpsTest):
+    """Add Juju user secret to model.
+
+    Args:
+        ops_test: PyTest object.
+
+    Returns:
+        secret ID of created secret.
+    """
+    juju_secret = await ops_test.model.add_secret(
+        name="worker-secrets", data_args=["juju-secret1=world", "juju-secret2=value2"]
+    )
+
+    secret_id = juju_secret.split(":")[-1]
+    return secret_id
+
+
+async def add_vault_secret(ops_test: OpsTest, path: str, key: str, value: str):
+    """Add Vault secret using charm action.
+
+    Args:
+        ops_test: PyTest object.
+        path: Path to write to.
+        key: Key to write secret to.
+        value: Secret value to write.
+    """
+    secret_added = False
+    for i in range(10):
+        action = (
+            await ops_test.model.applications[APP_NAME]
+            .units[0]
+            .run_action("add-vault-secret", path=path, key=key, value=value)
+        )
+        result = (await action.wait()).results
+        logger.info("add vault secret result: %s", result)
+        if "result" in result and result["result"] == "secret successfully created":
+            secret_added = True
+            break
+        time.sleep(2)
```

Comment:
Why do we need this? If the computer is slow, may we need 10 seconds?

Reply:
This is just to leave a small duration of time between each action attempt. The action generally works from the first attempt, but this is only here is a failsafe.